homework/index.js

```diff
@@ -30,13 +30,175 @@
+    const rightDiv = createAndAppend('div', contentDiv, { id: 'right-side' });
+
+    function getSelectedOption(sel) {
+      let opt;
+      for (let i = 0; i < sel.options.length; i++) {
+        opt = sel.options[i];
+        if (opt.selected === true) {
+          break;
+        }
+      }
+      return opt;
+    }
+
+    function getSelectedIndex(name, opt) {
+      let selectedIndex = 0;
+      for (let i = 0; i < name.length; i++) {
+        if (name[i].name === opt.text) {
+          selectedIndex = i;
+        }
+      }
+      return selectedIndex;
+    }
+
+    const allOptions = document.getElementById('select');
+    const firstSelOpt = getSelectedOption(allOptions);
+    const firstSelElemIndex = getSelectedIndex(repos, firstSelOpt);
+    const leftSideContent = createAndAppend('div', leftDiv, { id: 'left-side-content' });
+    const table = createAndAppend('table', leftSideContent, {});
+    const tr1 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr1, { text: 'Repository: ' });
+    const td2 = createAndAppend('td', tr1, {});
+    createAndAppend('a', td2, {
+      id: 'repo-name',
+      text: `${firstSelOpt.text}`,
+      href: `${repos[firstSelElemIndex].html_url}`,
+      target: '_blank',
+    });
+    const tr2 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr2, { text: 'Description: ' });
+    createAndAppend('td', tr2, {
+      id: 'repo-description',
+      text: `${repos[firstSelElemIndex].description}`,
+    });
+    const tr3 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr3, { text: 'Fork: ' });
+    createAndAppend('td', tr3, { id: 'repo-fork', text: `${repos[firstSelElemIndex].forks}` });
+    const tr4 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr4, { text: 'Updated: ' });
+    createAndAppend('td', tr4, {
+      id: 'repo-updated',
+      text: `${repos[firstSelElemIndex].updated_at}`,
+    });
+
+    function createRightSideInfos(contJsonData) {
+      const conts = contJsonData;
+      createAndAppend('h3', rightDiv, { text: 'Contributors' });
+      conts.forEach((_cont, index) => {
+        const contsInfo = createAndAppend('div', rightDiv, { class: 'cont-info' });
+        const contsInfoLeft = createAndAppend('div', contsInfo, { class: 'cont-info-left' });
+        const contsInfoRight = createAndAppend('div', contsInfo, { class: 'cont-info-right' });
+        createAndAppend('img', contsInfoLeft, {
+          src: `${conts[index].avatar_url}`,
+          alt: `${conts[index].login} avatar`,
+        });
+        createAndAppend('a', contsInfoLeft, {
+          text: `${conts[index].login}`,
+          href: `${conts[index].html_url}`,
+          target: '_blank',
+        });
+        createAndAppend('p', contsInfoRight, { text: `${conts[index].contributions}` });
+      });
+    }
+
+    const CONT_URL = repos[firstSelElemIndex].contributors_url;
+    fetchJSON(CONT_URL, (err, data) => {
```

Comment:
You're performing a `fetchJSON` call *inside* of a function called `createHtml`. It's better to extract this call into its own method and call it on window.onload (i.e. at the bottom of the page).